### DISCUSSION TITLE
Extract URLs from singlefilez files

### DISCUSSION THREAD

**Discussion by ghbook** [gildas-lormeau/SingleFileZ#159] [Q&A] [answered]
I have bunch of .zip.html files downloaded using singlefilez web extension over the years.
I want to extract URLs (`href`) from these files via nodejs script to a txt file.
How to do it since these are not typical html files but zips.

**Comment by gildas-lormeau** [comment]
You have to unzip the zip.html files for example with zip.js, see https://github.com/gildas-lormeau/zip.js. The URL can be found in the `manifest.json` file located in the root folder of the zip file, see `originalUrl`.

**Comment by gildas-lormeau** [comment, marked as answer]
FYI, here is a simple solution for obtaining the URLs from the command line on Linux/WSL/macOS (with `unzip` and `jq` installed):

```sh
for name in *.zip.html; do echo -n "$name -> "; unzip -p $name manifest.json | jq -r ".originalUrl"; done
```

**Reply by ghbook** [reply]
Thanks, it works.
